x-pack/plugins/security/server/lib/authorization/check_privileges_dynamically.ts

```diff
@@ -0,0 +1,22 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+export function checkPrivilegesDynamicallyWithRequestFactory(
```

Comment:
It wasn't immediately clear to me what made `checkPrivilegesDynamically` different from `checkPrivileges` until I read the implementation. I don't have a suggestion at this point for a better name, but I'll come back here if I do. So ignore this for now!

Reply:
The naming is super awkward... I struggled with what to call this, as it seems like the general way in which we want to "checkPrivileges", except we need the `checkPrivileges.atSpace`/`checkPrivileges.atSpaces`/`checkPrivileges.globally` for use in the SpacesClient... If you have any suggestions, I'd love to change this around.